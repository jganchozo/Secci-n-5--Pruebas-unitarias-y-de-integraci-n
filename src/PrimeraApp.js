import React from 'react';
import PropTypes from 'prop-types';
;



export const PrimeraApp = ({data, subtitle = 'Subtitle'}) => {

    return (
        <>
            <h1>{data}</h1>
            {/* <pre>{ JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{subtitle}</p>
        </>
    );
};

PrimeraApp.propTypes = {
    data: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitle: 'Default Subtitle',
}