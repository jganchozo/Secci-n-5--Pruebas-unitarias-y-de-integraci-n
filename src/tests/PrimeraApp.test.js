import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { PrimeraApp } from '../PrimeraApp';

describe('Pruebas en PrimeraApp', () => {
  
//   test('debe de mostrar el mensaje "Hola, Jose"', () => {
    
//     const saludo = 'Hola, Jose';
//     const { getByText } = render(<PrimeraApp data={saludo}/>);

//     expect(getByText(saludo)).toBeInTheDocument();
//   });

test('should mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, Jose';
    const wrapper = shallow(<PrimeraApp data={saludo} />);

    expect(wrapper).toMatchSnapshot();
});

test('should el subtitulo enviado por props', () => {
  
    const saludo = 'Hola, Jose';
    const subtitulo = 'Soy un subtitulo';
    const wrapper = shallow(<PrimeraApp data={saludo} subtitle={subtitulo} />);

    const textoParrafo = wrapper.find('p').text();
    //console.log(textoParrafo);
    expect(textoParrafo).toBe(subtitulo);
    
});


  
});
