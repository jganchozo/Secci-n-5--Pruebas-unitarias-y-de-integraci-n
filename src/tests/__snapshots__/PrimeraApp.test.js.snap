// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp should mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Jose
  </h1>
  <p>
    Default Subtitle
  </p>
</Fragment>
`;
